web/ui/react-app/src/pages/graph/PanelList.tsx

```diff
@@ -1,219 +1,181 @@
+  const [timeDriftError, setTimeDriftError] = useState();
+  const [fetchMetricsError, setFetchMetricsError] = useState();
+  const [useLocalTime, setUseLocalTime] = useLocalStorage('use-local-time', false);
+  const [enableQueryHistory, setEnableQueryHistory] = useLocalStorage('enable-query-history', false);
+
+  useEffect(() => {
+    (async () => {
+      const metricsResponse = await fetch(`${pathPrefix}/api/v1/label/__name__/values`, { cache: 'no-store' });
+      if (metricsResponse.ok) {
+        const json = await metricsResponse.json();
+        setMetricNames(json.data);
+      } else {
+        setFetchMetricsError(
+          `Error fetching metrics list: Unexpected response status when fetching metric names: ${metricsResponse.statusText}`
+        );
+      }
 
-  render() {
-    const { metricNames, pastQueries, timeDriftError, fetchMetricsError, panels } = this.state;
-    const { pathPrefix } = this.props;
-    return (
-      <>
-        <Row className="mb-2">
-          <Checkbox
-            id="query-history-checkbox"
-            wrapperStyles={{ margin: '0 0 0 15px', alignSelf: 'center' }}
-            onChange={this.toggleQueryHistory}
-            defaultChecked={this.isHistoryEnabled()}
-          >
-            Enable query history
-          </Checkbox>
-          <Checkbox
-            id="use-local-time-checkbox"
-            wrapperStyles={{ margin: '0 0 0 15px', alignSelf: 'center' }}
-            onChange={this.toggleUseLocalTime}
-            defaultChecked={this.useLocalTime()}
-          >
-            Use local time
-          </Checkbox>
-        </Row>
-        <Row>
-          <Col>
-            {timeDriftError && (
-              <Alert color="danger">
-                <strong>Warning:</strong> Error fetching server time: {timeDriftError}
-              </Alert>
-            )}
-          </Col>
-        </Row>
-        <Row>
-          <Col>
-            {fetchMetricsError && (
-              <Alert color="danger">
-                <strong>Warning:</strong> Error fetching metrics list: {fetchMetricsError}
-              </Alert>
-            )}
-          </Col>
-        </Row>
-        {panels.map(({ id, options }) => (
-          <Panel
-            onExecuteQuery={this.handleExecuteQuery}
-            key={id}
-            options={options}
-            onOptionsChanged={opts => this.handleOptionsChanged(id, opts)}
-            useLocalTime={this.state.useLocalTime}
-            removePanel={() => this.removePanel(id)}
-            metricNames={metricNames}
-            pastQueries={pastQueries}
-            pathPrefix={pathPrefix}
-          />
-        ))}
-        <Button color="primary" className="add-panel-btn" onClick={this.addPanel}>
-          Add Panel
-        </Button>
-      </>
-    );
-  }
-}
+      const browserTime = new Date().getTime() / 1000;
+      const timeResponse = await fetch(`${pathPrefix}/api/v1/query?query=time()`, {
```

Comment:
IIUC by switching to async/await for two subsequent requests (metric names and time), the requests are now done in sequence instead of in parallel, so maybe the page load will take a tiny bit longer? Any special reason to switch to async/await instead of callbacks here, or just preference?